### ISSUE TITLE
ETA is unreliable because of deduplication

### ISSUE THREAD

**Issue by pille** [restic/restic#3397]
Output of `restic version`
--------------------------
restic 0.12.0 compiled with go1.14.7 on linux/amd64

How did you run restic exactly?
-------------------------------
restic backup /src

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
SFTP

Expected behavior
-----------------
when i run the backup-process repeatedly, most chunks are already at the target and don't need to be transferred again.
the ETA should reflect the time needed until completion.
it can do better, by taking the average transfer speed of the last time-frame, instead of the overall transfer speed. the time-frame-parameter can be used to increase accuracy. 

in case it is already known after scanning, which/how many chunks actually need to be transferred, this information could also be used to increase accuracy even more.

Actual behavior
---------------
when i run the backup-process repeatedly, most chunks are already at the target and don't need to be transferred again.
the ETA is calculated rather simple and doesn't take this advantage into account.
because the transfer seems to start incredibly fast (GB/s) the ETA starts quite low and is constantly and slowly increasing, without any benefit.

Steps to reproduce the behavior
-------------------------------
repeat a backup of little-changing source multiple times and watch ETA. compare it to actual wall time.

Do you have any idea what may have caused this?
-----------------------------------------------
simple algorithm (see above)

Do you have an idea how to solve the issue?
-------------------------------------------
more complex algorithm (see above)

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
i still trust restic ;)

**Comment by MichaelEischer** [comment]
There has already been some discussion regarding this problem in the forum: https://forum.restic.net/t/eta-is-wrong-when-resuming-an-upload/3396/11

**Comment by matta** [comment]
I went ahead and implemented a more complex ETA estimate. It is in principle a sliding window, but as I hinted at in my last message, I found the naive approaches don't work very well. E.g windows based on fixed-time or even fixed-bytes-transferred seem to hit pathological cases in my testing. The key observation I found was that restic's backup progress is bursty, at least with a rate limit set against Backblace B2.  I handle this in #3563 by insisting that the sliding window holds a wide enough time window _and_ enough bytes transferred _and_ enough distinct measurements (as defined by discrete reported progress at least one second apart). With this in place I observed relatively good ETA values computed across the variety of problem cases I uncovered (mostly occurring when most of a backup is duplicate data, and with a restrictive rate limit in place).
